pytest_logger/plugin.py

```diff
@@ -174,6 +179,14 @@ def add_loggers(self, loggers, stdout_level=logging.NOTSET, file_level=logging.N
         self._enabled = True
         self._loggers.append((loggers, _sanitize_level(stdout_level), _sanitize_level(file_level)))
 
+    def set_formatter(self, formatter):
```

Comment:
Setting formatter as object instead of type changes meaning of timestamps (time is counted from test session start, instead of from testcase start (please see general comment)).

Reply:
set_formatter became set_formatter_class since 666aa66.
That's the only function that actually checks the validity of the input as it is public API.